semantic/src/main/kotlin/tools/samt/semantic/SemanticModel.kt

```diff
@@ -0,0 +1,284 @@
+package tools.samt.semantic
+
+import tools.samt.common.DiagnosticController
+import tools.samt.common.Location
+import tools.samt.parser.*
+
+/*
+Goals of the semantic model:
+- Model the entire package structure of the project
+- Model the types and their relationships
+- Resolve all references to types
+- Resolve all references to their declarations in the AST
+*/
+
+class SemanticModelBuilder(
+    private val files: List<FileNode>,
+    private val controller: DiagnosticController,
+) {
+
+    // TODO: check for duplicate declarations
```

Comment:
Is this TODO still pending? Comment also applies to the other methods marked with TODO

Reply:
You are correct, these TODOs are resolved, I've removed the related TODOs